tjhsst/apps/clubs/views.py

```diff
@@ -49,16 +50,62 @@ def index(request):
     )
 
 
+def dashboard(request):
+    clubs = request.user.clubs_following.all()
+    announcements = Announcement.objects.filter(club__in=clubs).order_by("-post_time")
+
+    return render(
+        request, "clubs/dashboard.html", context={"clubs": clubs, "announcements": announcements}
+    )
+
+
 def show(request, club_url):
     club = get_object_or_404(Club, url=club_url)
 
     return render(
         request,
         "clubs/show.html",
-        {"club": club, "can_edit": request.user.is_superuser or request.user in club.admins.all()},
+        {
+            "club": club,
+            "can_edit": request.user.is_superuser or request.user in club.admins.all(),
+            "is_following": request.user in club.followers.all(),
+        },
     )
 
 
+def follow(request, club_url):
+    club = get_object_or_404(Club, url=club_url)
+
+    if request.user.is_authenticated:
+        if request.user in club.followers.all():
+            club.followers.remove(request.user)
+        else:
+            club.followers.add(request.user)
+        club.save()
+        return redirect("clubs:show", club.url)
+    else:
+        raise Http404
+
+
+def post(request, club_url):
+    club = get_object_or_404(Club, url=club_url)
+
+    if request.user.is_superuser or request.user in club.admins.all():
+        if request.method == "POST":
+            form = AnnouncementCreationForm(request.POST, initial={"club": club})
+            if form.is_valid():
+                post = form.save()
+                post.club = club
+                post.save()
+                return redirect("clubs:show", club.url)
+        else:
+            form = AnnouncementCreationForm(initial={"club": club})
+
```

Comment:
I think, without this, an unauthenticated user would raise an exception?

Or, add `@login_required`?

```suggestion
    else:
        form = None
```